Secure baselines fails on first run

## Expected Behavior

Baselines not to fail

## Actual Behavior

Running secure baselines for the first time on an account gives the following error - 

```
module.baselines.module.cloudtrail.module.cloudtrail-log-bucket.data.aws_iam_policy_document.default: Read complete after 0s [id=<REDACTED>]
module.baselines.module.cloudtrail.module.cloudtrail-bucket.aws_s3_bucket_policy.replication: Creation complete after 0s [id=<REDACTED>]
module.baselines.module.cloudtrail.module.cloudtrail-log-bucket.aws_s3_bucket_public_access_block.default: Creation complete after 1s [id=<REDACTED>]
module.baselines.module.cloudtrail.module.cloudtrail-log-bucket.aws_s3_bucket_policy.default: Creating...
module.baselines.module.cloudtrail.module.cloudtrail-log-bucket.aws_s3_bucket_policy.default: Creation complete after 0s [id=<REDACTED>]
module.baselines.module.cloudtrail.module.cloudtrail-bucket.aws_s3_bucket.default: Creation complete after 9s [id=<REDACTED>]
module.baselines.module.cloudtrail.data.aws_iam_policy_document.cloudtrail: Reading...
module.baselines.module.s3-replication-role.data.aws_iam_policy_document.default-policy: Reading...
module.baselines.module.config-eu-west-2["enabled"].aws_config_config_rule.multi-region-cloudtrail-enabled[0]: Creating...
module.baselines.module.cloudtrail.module.cloudtrail-bucket.aws_s3_bucket_public_access_block.default: Creating...
module.baselines.module.config-eu-west-2["enabled"].aws_config_config_rule.cloudtrail-enabled[0]: Creating...
module.baselines.module.s3-replication-role.data.aws_iam_policy_document.default-policy: Read complete after 0s [id=<REDACTED>]
module.baselines.module.cloudtrail.data.aws_iam_policy_document.cloudtrail: Read complete after 0s [id=<REDACTED>]
module.baselines.module.cloudtrail.module.cloudtrail-bucket.data.aws_iam_policy_document.default: Reading...
module.baselines.module.s3-replication-role.aws_iam_policy.default: Creating...
module.baselines.module.cloudtrail.module.cloudtrail-bucket.data.aws_iam_policy_document.default: Read complete after 0s [id=<REDACTED>]
module.baselines.module.s3-replication-role.aws_iam_policy.default: Creation complete after 1s [id=<REDACTED>]
module.baselines.module.s3-replication-role.aws_iam_role_policy_attachment.default: Creating...
module.baselines.module.s3-replication-role.aws_iam_role_policy_attachment.default: Creation complete after 0s [id=<REDACTED>]
module.baselines.module.cloudtrail.module.cloudtrail-bucket.aws_s3_bucket_public_access_block.default: Creation complete after 1s [id=<REDACTED>]

Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.unauthorised-api-calls,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 54, in resource "aws_cloudwatch_log_metric_filter" "unauthorised-api-calls":
  54: resource "aws_cloudwatch_log_metric_filter" "unauthorised-api-calls" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.sign-in-without-mfa,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 84, in resource "aws_cloudwatch_log_metric_filter" "sign-in-without-mfa":
  84: resource "aws_cloudwatch_log_metric_filter" "sign-in-without-mfa" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.root-account-usage,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 115, in resource "aws_cloudwatch_log_metric_filter" "root-account-usage":
 115: resource "aws_cloudwatch_log_metric_filter" "root-account-usage" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.iam-policy-changes,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 145, in resource "aws_cloudwatch_log_metric_filter" "iam-policy-changes":
 145: resource "aws_cloudwatch_log_metric_filter" "iam-policy-changes" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.cloudtrail-configuration-changes,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 175, in resource "aws_cloudwatch_log_metric_filter" "cloudtrail-configuration-changes":
 175: resource "aws_cloudwatch_log_metric_filter" "cloudtrail-configuration-changes" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.sign-in-failures,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 205, in resource "aws_cloudwatch_log_metric_filter" "sign-in-failures":
 205: resource "aws_cloudwatch_log_metric_filter" "sign-in-failures" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.cmk-removal,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 235, in resource "aws_cloudwatch_log_metric_filter" "cmk-removal":
 235: resource "aws_cloudwatch_log_metric_filter" "cmk-removal" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.s3-bucket-policy-changes,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 265, in resource "aws_cloudwatch_log_metric_filter" "s3-bucket-policy-changes":
 265: resource "aws_cloudwatch_log_metric_filter" "s3-bucket-policy-changes" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.config-configuration-changes,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 295, in resource "aws_cloudwatch_log_metric_filter" "config-configuration-changes":
 295: resource "aws_cloudwatch_log_metric_filter" "config-configuration-changes" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.security-group-changes,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 325, in resource "aws_cloudwatch_log_metric_filter" "security-group-changes":
 325: resource "aws_cloudwatch_log_metric_filter" "security-group-changes" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.nacl-changes,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 355, in resource "aws_cloudwatch_log_metric_filter" "nacl-changes":
 355: resource "aws_cloudwatch_log_metric_filter" "nacl-changes" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.network-gateway-changes,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 385, in resource "aws_cloudwatch_log_metric_filter" "network-gateway-changes":
 385: resource "aws_cloudwatch_log_metric_filter" "network-gateway-changes" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.route-table-changes,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 415, in resource "aws_cloudwatch_log_metric_filter" "route-table-changes":
 415: resource "aws_cloudwatch_log_metric_filter" "route-table-changes" {


Error: Creating/Updating CloudWatch Log Metric Filter failed: ResourceNotFoundException: The specified log group does not exist.

  with module.baselines.module.securityhub-alarms.aws_cloudwatch_log_metric_filter.vpc-changes,
  on .terraform/modules/baselines/modules/securityhub-alarms/main.tf line 445, in resource "aws_cloudwatch_log_metric_filter" "vpc-changes":
 445: resource "aws_cloudwatch_log_metric_filter" "vpc-changes" {

module.baselines.module.cloudtrail.module.cloudtrail-bucket.aws_s3_bucket_policy.default: Creating...
module.baselines.module.config-eu-west-2["enabled"].aws_config_config_rule.multi-region-cloudtrail-enabled[0]: Creation complete after 1s [id=<REDACTED>]
module.baselines.module.cloudtrail.module.cloudtrail-bucket.aws_s3_bucket_policy.default: Creation complete after 1s [id=<REDACTED>]
module.baselines.module.cloudtrail.aws_cloudtrail.cloudtrail: Creating...
module.baselines.module.config-eu-west-2["enabled"].aws_config_config_rule.cloudtrail-enabled[0]: Creation complete after 2s [id=<REDACTED>]
module.baselines.data.aws_iam_policy_document.config-publish-policy: Reading...
module.baselines.data.aws_iam_policy_document.config-publish-policy: Read complete after 0s [id=<REDACTED>]
module.baselines.aws_iam_policy.config-publish-policy: Creating...
module.baselines.aws_iam_policy.config-publish-policy: Creation complete after 1s [id=<REDACTED>]
module.baselines.aws_iam_role_policy_attachment.config-publish-policy: Creating...
module.baselines.aws_iam_role_policy_attachment.config-publish-policy: Creation complete after 0s [id=<REDACTED>]
module.baselines.module.cloudtrail.aws_cloudtrail.cloudtrail: Creation complete after 3s [id=<REDACTED>]
Error: Process completed with exit code 1.
```

## Steps to Reproduce the Problem

  1. Create a new account and allow the baselines to run
  1.
  1.

## Other information

The scheduled baseline job is running successfully, it looks like this is a dependancy issue in the module terraform.

--- a/terraform/environments/bootstrap/secure-baselines/main.tf
+++ b/terraform/environments/bootstrap/secure-baselines/main.tf
@@ -1,5 +1,5 @@
 module "baselines" {
-  source = "github.com/ministryofjustice/modernisation-platform-terraform-baselines?ref=v3.0.0"
+  source = "github.com/ministryofjustice/modernisation-platform-terraform-baselines?ref=v3.0.1"
   providers = {
     # Default and replication regions
     aws                    = aws.workspace-eu-west-2

